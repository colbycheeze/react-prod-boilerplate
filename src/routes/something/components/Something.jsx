import React, { PureComponent } from 'react';
import PT from 'prop-types';
import { connect } from 'react-redux';
import { somethingSelector, doSomething } from '../modules/something';

const mapStateToProps = (state) => ({
  thing: somethingSelector(state).thing,
});
const actions = {
  doSomething,
};
export class Something extends PureComponent {
  static propTypes = {
    thing: PT.string.isRequired,
    doSomething: PT.func.isRequired,
  }

  handleClick = () => {
    this.props.doSomething();
  }

  render = () => {
    const { thing } = this.props;

    return (
      <div>
        Something: {thing}
        <br />
        <button onClick={this.handleClick}>Do Something</button>
      </div>
    );
  }
}
export default connect(mapStateToProps, actions)(Something);
