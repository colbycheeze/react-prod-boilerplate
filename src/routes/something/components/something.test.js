import React from 'react';
import { Something } from './Something';

const setup = () => {
  const spies = {
    doSomething: jest.fn(),
  };

  const props = {
    thing: 'Test Thing',
    ...spies,
  };

  return { spies, props };
};

describe('<Something />', () => {
  it('Shows a default thing on load', () => {
    const { props } = setup();
    const wrapper = shallow(<Something {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Does something when the button is clicked', () => {
    const { props, spies } = setup();
    const wrapper = shallow(<Something {...props} />);

    expect(spies.doSomething).not.toBeCalled();
    wrapper.find('button').simulate('click');
    expect(spies.doSomething).toBeCalled();
  });
});
