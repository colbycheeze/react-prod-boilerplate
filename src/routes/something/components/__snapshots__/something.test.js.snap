exports[`<Something /> Shows a default thing on load 1`] = `
<div>
  Something: 
  Test Thing
  <br />
  <button
    onClick={[Function]}>
    Do Something
  </button>
</div>
`;
